org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BundleUtils.java

```diff
@@ -194,7 +215,9 @@ private static void refreshBundles(Set<Bundle> toRefresh, FrameworkWiring framew
 			frameworkWiring.refreshBundles(toRefresh, new FrameworkListener() {
 				@Override
 				public void frameworkEvent(FrameworkEvent event) {
-					latch.countDown();
+					if (event.getType() == FrameworkEvent.PACKAGES_REFRESHED) {

```

Comment:
when it's FrameworkEvent.ERROR, also exit?

Reply:
I tried to find if there is any example also handling other events when refresh the packages but with no success.

Is there any example I can refer to?

Reply:
No example yet. i just read the API document of FrameworkListener, it says when there is any exception happening during refresh, it will report ERROR event.

Reply:
I see. Then how about log an error message when this happens?

Reply:
if we always are able to receive a FrameworkEvent.PACKAGES_REFRESHED event regardless of success or fail, i'm OK. My point is to make sure the latch.await is not blocking when there is bad status happening.